app/models/document_bundle.rb

```diff
@@ -1,6 +1,10 @@
 class DocumentBundle < ApplicationRecord
   belongs_to :crime_application
-  has_many :documents, dependent: :destroy
+  has_many :documents, dependent: :destroy do
```

Comment:
Maybe this is simpler?

```ruby
has_many :documents, dependent: :destroy
has_many :stored_documents, -> { where.not(s3_object_key: nil) }, class_name: 'Document'
```

We would need both, as there are use-cases where we want all documents to be returned, including those not persisted to S3 (for example in the upload page when reloading in non-JS version, to see errors).

It also avoids adding a method to the `Document` model, and might be easier to stub.

Reply:
In addition to the above comment, there is another way, I think the simplest one, which is to add a scope to the `Document` model:

```ruby
scope :stored, -> { where.not(s3_object_key: nil) }
```

That way we can do:

```ruby
Document.stored

bundle = DocumentBundle.last
bundle.documents.stored
```

In any case at some point we will need a refactor to remove the concept of bundle as it seems will not longer be needed 😄

Reply:
That's much cleaner 🤩 I've made those changes

Reply:
Nice! 🥳